staging/src/k8s.io/apiserver/pkg/server/config_test.go

```diff
@@ -32,20 +32,58 @@ import (
 	"k8s.io/client-go/rest"
 )
 
-func TestAuthorizeClientBearerToken(t *testing.T) {
+func TestAuthorizeClientBearerTokenNoops(t *testing.T) {
 	// All of these should do nothing (not panic)
-	AuthorizeClientBearerToken(nil, nil, nil)
-	AuthorizeClientBearerToken(nil, nil, &AuthorizationInfo{})
-	AuthorizeClientBearerToken(nil, &AuthenticationInfo{}, nil)
-	AuthorizeClientBearerToken(nil, &AuthenticationInfo{}, &AuthorizationInfo{})
-	AuthorizeClientBearerToken(&rest.Config{}, nil, nil)
-	AuthorizeClientBearerToken(&rest.Config{}, nil, &AuthorizationInfo{})
-	AuthorizeClientBearerToken(&rest.Config{}, &AuthenticationInfo{}, nil)
-	AuthorizeClientBearerToken(&rest.Config{}, &AuthenticationInfo{}, &AuthorizationInfo{})
-	AuthorizeClientBearerToken(&rest.Config{BearerToken: "mu"}, nil, nil)
-	AuthorizeClientBearerToken(&rest.Config{BearerToken: "mu"}, nil, &AuthorizationInfo{})
-	AuthorizeClientBearerToken(&rest.Config{BearerToken: "mu"}, &AuthenticationInfo{}, nil)
-	AuthorizeClientBearerToken(&rest.Config{BearerToken: "mu"}, &AuthenticationInfo{}, &AuthorizationInfo{})
+	for il := 0; il <= 2; il++ {
```

Comment:
Sorry, I mean't something more along the lines of
```go
       restCfgs := []*rest.Config{nil, &rest.Config{}, &rest.Config{BearerToken: "mu"}}                                                                             
       authnInfos := []*AuthenticationInfo{nil, &AuthenticationInfo{}}
       authzInfos := []*AuthorizationInfo{nil, &AuthorizationInfo{}}
       for _, restCfg := range restCfgs {
               for _, authnInfo := range authnInfos {
                       for _, authzInfo := range authzInfos {
                               AuthorizeClientBearerToken(restCfg, authnInfo, authzInfo)                                                                            
                        }
                }
        }
```

Reply:
Yeah, I considered that, but decided against it on the grounds that the function being called has the potential to side-effect the arguments so it is better to give it, in the non-nil cases, pointers to freshly initialized memory.  And as long as I am doing that, I might as well test that there were no side effects.

Reply:
Do we check that function arguments are const anywhere else? This change makes the test pretty hard for me to parse. I'm not sure the trade-off is worth it.

Reply:
> on the grounds that the function being called has the potential to side-effect the arguments so it is better to give it, in the non-nil cases, pointers to freshly initialized memory

we can do that with generating functions. I agree the new form is harder to read

Reply:
I revised the test to use generating functions; hopefully it is easier to understand now.